### PR TITLE
RC doc updates

### PR DESCRIPTION
**Pull request by BoD**
Documentation updates for the next release.
To be merged when the RC is about to be released.

### CODE REVIEW

File: docs/source/advanced/operation-variables.mdx [hunk 1]
```diff
@@ -21,46 +23,55 @@ query GetTodos($first: Int, $offset: Int) {
...
+val query = GetTodosQuery(null, null)
 // Send explicit values for both variables
-val query = GetTodosQuery(Optional.Present(100), Optional.Present(0))
+val query = GetTodosQuery(100, 0)
 ```
 
-You can also opt in to optional variables globally in your Gradle config:
+If you pass `null` as the value for either of these variables, Apollo Android sends `null` to the server _as the value of that variable_ instead of omitting it entirely.
+
+You can also opt out to optional variables globally in your Gradle config:
```
[the remark this answers is not in the document]
Comment (martinbonnin) [reply docs/source/advanced/operation-variables.mdx:70]: Also probably invert the logic ? 

### Making nullable variables non-optional by default

```
generateOptionalOperationVariables.set(false)
```

For exception, you can still use `@optional(if: true)`

**Comment by martinbonnin** [comment]
👍  Reminder to remove the corresponding section in the migration guide

File: docs/source/advanced/operation-variables.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinbonnin) [reply docs/source/advanced/operation-variables.mdx:70]: Sorry, I meant that the value of adding `@optional(if: true)` for the exceptional cases that explicitly require optional variables far outweight the value of adding `@optional(if: false)` in the 98% of the cases.

So I wouldn't even describe how to use `@optional(if: false)` but instead focus on the `@optional(if: true)` cases

[the remark this answers is not in the document]
Comment (BoD) [reply docs/source/advanced/operation-variables.mdx:70]: Ah got you! Makes sense indeed.

[the remark this answers is not in the document]
Comment (martinbonnin) [reply docs/source/advanced/operation-variables.mdx:70]: Don't know if that makes a lot more sense 😅 . Basically remove `@optional(if: false)`, only talk about `@optional(if: true)`

File: docs/source/migration/3.0.mdx [hunk 2]
```diff
@@ -549,9 +549,9 @@ Input.optional(value)
 Optional.presentIfNotNull(value)
 ```
 
-#### Non-optional variables by default
+#### Non-optional variables generation
 
-By default, Apollo Android 3.x represents operation variables that _don't_ have a default value as non-optional Kotlin constructor parameters.
+As with 2.x, Apollo Android 3.x represents operation variables as optional parameters.
```
Comment (martinbonnin) [inline docs/source/migration/3.0.mdx:554]: I'd remove that section altogether. If it didn't change, it doesn't belong in the migration guide

**Pull request merged**